go-dlv.el

```diff
@@ -131,6 +134,27 @@ and source-file directory for your debugger."
   (setq paragraph-start comint-prompt-regexp)
   (run-hooks 'go-dlv-mode-hook))
 
+;;;###autoload
+(defun dlv-current-func ()
+  "Debug the current program or test stopping at the beginning of the current function."
+  (interactive)
+  (setq test-name nil)
+  (ignore-errors (setq test-name (go-test-get-current-test)))
```

Comment:
Unfortunately `go-test-get-current-test` calls `error` if test function cannot be found. I am not sure how to avoid `ignore-errors` here without re-implementing `go-test-get-current-test`